streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -582,6 +620,9 @@ object WriteAheadLogSuite {
       allowBatching: Boolean): Seq[String] = {
     val wal = createWriteAheadLog(logDirectory, closeFileAfterWrite, allowBatching)
     val data = wal.readAll().asScala.map(byteBufferToString).toSeq
+    // The thread pool for parallel recovery gets killed with wal.close(). Therefore we need to
+    // eagerly compute data, otherwise the lazy computation will fail.
+    data.length
```

Comment:
Could you just change `toSeq` to `toArray`? `toArray` will drain the Iterator at once.